Tests/Tests/SDWebCacheCategoriesTests.m

```diff
@@ -143,4 +179,105 @@ - (void)testUIViewImageProgressKVOWork {
+    imageView.sd_imageTransition = SDWebImageTransition.fadeTransition;
+    imageView.sd_imageTransition.duration = 1;
+    
+#if SD_UIKIT
+    [self.window addSubview:imageView];
+#else
+    imageView.wantsLayer = YES;
+    [self.window.contentView addSubview:imageView];
+#endif
+    
+    UIImage *placeholder = [[UIImage alloc] initWithContentsOfFile:[self testJPEGPath]];
+    
+    NSURL *originalImageURL = [NSURL URLWithString:kTestJpegURL];
+    __weak typeof(imageView) wimageView = imageView;
+    [imageView sd_setImageWithURL:originalImageURL
+                 placeholderImage:placeholder
+                          options:SDWebImageForceTransition
+                        completed:^(UIImage * _Nullable image, NSError * _Nullable error, SDImageCacheType cacheType, NSURL * _Nullable imageURL) {
+                            __strong typeof(wimageView) simageView = imageView;
+                            // Delay to let CALayer commit the transition in next runloop
+                            dispatch_after(dispatch_time(DISPATCH_TIME_NOW, kMinDelayNanosecond), dispatch_get_main_queue(), ^{
+                                // Check current view contains layer animation
+                                NSArray *animationKeys = simageView.layer.animationKeys;
+                                expect(animationKeys.count).beGreaterThan(0);
+                                [expectation fulfill];
+                            });
+                        }];
+    
+    [self waitForExpectationsWithCommonTimeout];
+}
+
+- (void)testUIViewIndicatorWork {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"UIView indicator does not work"];
+    
+    UIImageView *imageView = [[UIImageView alloc] init];
+    imageView.sd_imageIndicator = SDWebImageActivityIndicator.grayLargeIndicator;
+    // Test setter trigger removeFromSuperView
+    imageView.sd_imageIndicator = SDWebImageProgressIndicator.defaultIndicator;
+    
+    NSURL *originalImageURL = [NSURL URLWithString:kTestJpegURL];
+    __weak typeof(imageView) wimageView = imageView;
+    [imageView sd_setImageWithURL:originalImageURL
+                 placeholderImage:nil options:SDWebImageFromLoaderOnly progress:^(NSInteger receivedSize, NSInteger expectedSize, NSURL * _Nullable targetURL) {
+                     dispatch_async(dispatch_get_main_queue(), ^{
+                         __strong typeof(wimageView) simageView = imageView;
+                         UIView *indicatorView = simageView.subviews.firstObject;
+                         expect(indicatorView).equal(simageView.sd_imageIndicator.indicatorView);
+                         
+                         if (receivedSize <= 0 || expectedSize <= 0) {
+                             return;
+                         }
+                         
+                         // Current progess indicator update is called after progressBlock
```

Comment:
Opps...This comment should be removed because of that #2437 change the order so that this can pass